xbmc/filesystem/FavouritesDirectory.cpp

```diff
@@ -197,8 +197,15 @@ std::string CFavouritesDirectory::GetExecutePath(const CFileItem &item, const st
       execute = StringUtils::Format("ActivateWindow(%s,%s,return)", contextWindow.c_str(), StringUtils::Paramify(item.GetPath()).c_str());
   }
   //! @todo STRING_CLEANUP
-  else if (item.IsScript() && item.GetPath().size() > 9) // plugin://<foo>
+  else if (StringUtils::StartsWith(item.GetPath(), "script://"))
```

Comment:
This is not identical because this also allows to call the builtin with just `script://` which will not result in anything useful. The same applies for the other cases you added.

Furthermore I'd prefer to keep using `item.IsScript()` because that way we don't clutter the code with static `script://` strings all over the place.